Modals/Sources/Logging/ModalPresentationWillTransitionLogEvent.swift

```diff
@@ -0,0 +1,87 @@
+import Logging
+import UIKit
+
+#if !os(watchOS)
+/// Strutured log event metadata that represents a modal presentation transition is about to happen.
+///
+/// Log consumers can identify this event from the ``ModalsLogging/defaultLoggerLabel`` logger and
+/// unmarshal it using the ``init(metadata:)`` initializer.
+public struct ModalPresentationWillTransitionLogEvent: CustomStringConvertible {
+
+    public static let eventType = "\(ModalsLogging.defaultLoggerLabel).willtransition"
```

Comment:
nit: assuming from other strings you want this camelcased?
```suggestion
    public static let eventType = "\(ModalsLogging.defaultLoggerLabel).willTransition"
```

Reply:
I don't think it matters much, but the string it's copied from in Market is also lowercased like this.